### PR TITLE
[Feature, Documentation, DEV-43] very rough first cut of contracts

### PR DESCRIPTION
**Pull request by davesag**
Updated some of the docs too.

*Note ready to be merged*

### CODE REVIEW

File: notes.md [hunk 1]
```diff
@@ -51,6 +51,10 @@
 * The total number of shares outstanding is the difference between a cap and `token.totalSupply()` and is rightly a function of the `CrowdsaleContract`.
 * The date the share was purchased is a property of the block the purchase was processed in so there is no need to store it in the token. However the user database might wish to record that information as well for convenience.
 * The price the user paid is a function of the ETH price per Token (which can change during the crowdsale) and ETH price at that point (which can also be set in the `CrowdsaleContract`) and the number of ETH spent, which the `CrowdsaleContract` also knows and which is returned in the transaction receipt.
+* Tokens can only be minted in whole numbers so `floor` the conversion and put the remainder, less some gas, into a `RefundVault`.
+* Move the ETH to be turned into Tokens into the Crowdsale's `wallet`.
```
Comment (tavakyan) [inline notes.md:55]: Should* we configure the crowdsale wallet to forward to a multisig wallet controlled by cofounders to protect it? Like this: https://blog.zeppelin.solutions/handling-crowdsale-funds-securely-c348467da74c

"We propose a simple solution: timelock the funds in an intermediate contract until the crowdsale is over. When the crowdsale is finalized, funds can be sent to the project’s multisig wallet."

File: notes.md [hunk 2]
```diff
@@ -51,6 +51,10 @@
...
+* Move the remainder into a `RefundVault` called `refundWallet`.
+* People with refunds can claim their refunds through the Crowdsale Website.
```
Comment (tavakyan) [inline notes.md:57]: Are refunds typically something that are given automatically during the crowdsale or after the crowdsale is complete?

**Review by tavakyan — APPROVED** [review]

File: contracts/CTKNCrowdsale.sol [hunk 3]
```diff
@@ -1,90 +1,31 @@
...
-
-    function Crowdsale(
-        uint256 _cap,
-        uint256 _goal,
-        uint256 _startTime,
-        uint256 _endTime,
+    function CTKNCrowdsale(
+        uint256 _openingTime,
+        uint256 _closingTime,
         uint256 _rate,
```
Comment (tavakyan) [inline contracts/CTKNCrowdsale.sol]: Just to clarify is this the rate at which we will buy/convert eth to usd for the $54 share price at end of crowdsale?  So if we say we'll buy eth @ $1080 USD and so 20 shares per eth per end of crowdsale?

**Review by tavakyan — DISMISSED** [review]

File: notes.md (hunk 1 above)
Comment (tavakyan) [reply notes.md:55]: Looks like in the recent implementation of the crowdsale token they just forward it to the beneficiary.  Not sure I understood why that article recommended a timelock; seems unnecessary

Comment (davesag) [reply notes.md:55]: I've changed this now to base out crowdsale on the following base contracts

* `CappedCrowdsale`,
* `RefundableCrowdsale`, and
* `MintedCrowdsale`

I need to make a number of overrides to that base however to suit our needs.

File: notes.md (hunk 2 above)
Comment (davesag) [reply notes.md:57]: After discussion with the lawyer we'll release Tokens at the end of the crowdsale

**Pull request merged**